src/staticMethods/bindClickHandler.js

```diff
@@ -0,0 +1,25 @@
+let bodyClickListenerAdded = false
+const clickHandlers = {}
+
+export function bindClickHandler (attr = 'data-swal-template') {
+  clickHandlers[attr] = this
+
+  if (!bodyClickListenerAdded) {
+    document.body.addEventListener('click', bodyClickListener)
+    bodyClickListenerAdded = true
+  }
+}
+
+const bodyClickListener = (event) => {
+  // 1. using .parentNode instead of event.path because of better support by old browsers https://stackoverflow.com/a/39245638
+  // 2. using .parentNode instead of .parentElement because of IE11 + SVG https://stackoverflow.com/a/36270354
+  for (let el = event.target; el.parentNode !== document; el = el.parentNode) {
```

Comment:
Hope the failing test doesn't detract from the point that we can iterate through parent elements with a for loop.
I missed the "Notes" section in [MDN `node.parentNode` docs](https://developer.mozilla.org/en-US/docs/Web/API/Node/parentNode).
Test is failing because the element receiving the click event (`buttonTriggerPopup`) is not attached to `document` tree when the click event is processed.

Also the condition should have been `el !== document` instead of `el.parentElement !== document` in the first place, to include the final `<html>` element in the iteration.

```suggestion
  for (let el = event.target; el && el !== document; el = el.parentNode) {
```

Reply:
Done, tests are passing now 👍